benchmarks/graphics_pipeline/main.cpp

```diff
@@ -783,6 +850,60 @@ void ProjApp::Setup()
     }
 }
 
+void ProjApp::KeyDown(ppx::KeyCode key)
+{
+    mPressedKeys.insert(key);
+}
+
+void ProjApp::KeyUp(ppx::KeyCode key)
+{
+    mPressedKeys.erase(key);
+}
+
+void Person::Move(MovementDirection dir, float distance)
+{
+    if (dir == MovementDirection::FORWARD) {
+        location += float3(0, 0, distance);
+    }
+    else if (dir == MovementDirection::LEFT) {
+        location += float3(distance, 0, 0);
+    }
+    else if (dir == MovementDirection::RIGHT) {
+        location += float3(-distance, 0, 0);
+    }
+    else if (dir == MovementDirection::BACKWARD) {
+        location += float3(0, 0, -distance);
+    }
+    else {
+        PPX_ASSERT_MSG(false, "unhandled direction code " << static_cast<int>(dir));
```

Comment:
if you use a switch, the compiler will warn if it's not exhaustive, meaning no need for this assert.

Reply:
Done.